Jamfile

```diff
@@ -237,6 +237,7 @@ rule warnings ( properties * )
 		result += <cxxflags>-Wno-c++98-compat-pedantic ;
 		result += <cflags>-Wno-padded ;
 		result += <cflags>-Wno-global-constructors ;
+		result += <cflags>-Wheader-hygiene ;
```

Comment:
hm.. yeah, that makes sense. I tried building with this warning enabled specifically yesterday, and I saw it fire in the tests. but I realize now that the tests were never built with all the warnings enabled to begin with